wandb/data_types.py

```diff
@@ -965,12 +982,26 @@ def captions(cls, audio_list):
         else:
             return ["" if c is None else c for c in captions]
 
+    def resolve_ref(self):
+        if Audio.path_is_reference(self._path):
+            # this object was already created using a ref:
+            return self._path
+
+        source_artifact = self.artifact_source["artifact"]
+        return source_artifact.get_path(
+            self._path.split("./artifacts/" + source_artifact.name + "/")[-1]
+        ).ref()
```

Comment:
I think this might be more stable if written as: `source_artifact.manifest.get_entry_by_path(self._path).ref`